Automattic/jetpack
# Jetpack AI: fair usage quota handling on the UI, part 1

## Proposed changes:


* Handle the fair usage limit messaging on some points of the UI:
   * AI Assistant block
   * AI Excerpt generator
   * AI Image Generator and AI Featured Image Generator
* Rename the `UpgradePrompt` component to `QuotaExceededMessage` because it may hold an upgrade message but also the fair usage notice
* Fix image generation modal to disable the UI when the site requires an upgrade
* Notes:
   * this PR does not handle the messaging on the extensions (the lines under the AI Control component); I will address it on a follow up 
   * the message being displayed is subject to change as per design project: p1HpG7-tUE-p2

### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?



## Testing instructions:






* Test it in pair with D159074-code
* Make sure you sandbox the `public-api` domain
* On the `0-sandbox.php` file of your sandbox, set the following filters:

```
add_filter( 'jetpack_ai_tier_licensed_quantity', function() { return 1; } );
add_filter( 'jetpack_ai_current_period_requests_count', function() { return 3001; } );
```

* Go to the block editor
* Add an AI Assistant block; confirm the UI is blocked and the fair usage message is displayed:

<img width="500" alt="Screenshot 2024-08-23 at 16 28 47" src="https://github.com/user-attachments/assets/947e57fe-d65f-4277-96d8-38c81437fd96">

* Add an Image block and click the "Generate with AI"; confirm the UI is blocked and the fair usage message is displayed:

<img width="500" alt="Screenshot 2024-08-23 at 16 31 56" src="https://github.com/user-attachments/assets/2977b2dc-0921-45a8-ad2e-a117e56b63c1">

* Check the Excerpt section on the post sidebar; confirm the UI related to AI generation is blocked and the fair usage message is displayed:

<img width="250" alt="Screenshot 2024-08-23 at 16 28 00" src="https://github.com/user-attachments/assets/31304e1b-5dbd-4ee3-bef2-a19737191532">

* On the Jetpack sidebar, AI Assistant section, confirm the features are disabled:

<img width="250" alt="Screenshot 2024-08-23 at 16 28 19" src="https://github.com/user-attachments/assets/f22c0d5b-2be7-498f-accd-cdd22c86f8b3">

* Change the filters on the `0-sandbox.php` to:

```
add_filter( 'jetpack_ai_tier_licensed_quantity', function() { return 1; } );
add_filter( 'jetpack_ai_current_period_requests_count', function() { return 2999; } );
```

* Check the UI elements again and confirm they are now enabled, and the fair usage message is gone
